cdci_data_analysis/analysis/drupal_helper.py

```diff
@@ -42,6 +42,51 @@ def analyze_drupal_output(drupal_output, operation_performed=None):
         return drupal_output.json()
 
 
+# TODO extend to support the sending of the requests also in other formats besides hal_json
+# not necessary at the moment, but perhaps in the future it will be
+def execute_drupal_request(url,
+                           params=None,
+                           data=None,
+                           method='get',
+                           headers=None,
+                           files=None,
+                           request_format='hal_json',
+                           sentry_client=None):
+    try:
+        if method == 'get':
+            if params is None:
+                params = {}
+            params['_format'] = request_format
+            return requests.get(url,
+                                params={**params},
+                                headers=headers)
+        elif method == 'post':
+            if data is None:
+                data = {}
+            if params is None:
+                params = {}
+            params['_format'] = request_format
+            return requests.post(url,
+                                 params={**params},
+                                 data=data,
+                                 files=files,
+                                 headers=headers
+                                 )
+    except Exception as e:
+        logger.warning(f"an issue occurred when performing a request to the product gallery, "
+                       f"this is likely to be a connection related problem, we are investigating and "
+                       f"try to solve it as soon as possible")
+        if sentry_client is not None:
+            sentry_client.capture('raven.events.Message',
+                                  message=f'exception when performing a request to the product gallery: {repr(e)}')
+        else:
+            logger.warning("sentry not used")
+
+        raise BadRequest('issue when performing a request to the product gallery',
```

Comment:
`BadRequest` should mean user did something wrong, and should modify the requet. Some InternalError would be better here.